### PR TITLE
Add "list" command to show all available stacks

### PR DESCRIPTION
**Pull request by kainosnoema**
An open question is whether this command should show stacks already in CF (like show does), or just the local stacks. Or maybe both, with status of creation? Currently this just lists local stacks.

screen shot 2018-08-02 at 11 26 01 am
screen shot 2018-08-02 at 3 45 42 pm

### CODE REVIEW

File: backend/config.go [hunk 1]
```diff
@@ -86,6 +95,14 @@ func (s *configStore) resolveStack(path string, st stackConfig) stackConfig {
 			stack.Region = c.Defaults.Region
 		}
 
+		if stack.Parameters == nil {
+			stack.Parameters = map[string]interface{}{}
```
Comment (kainosnoema) [inline backend/config.go:99]: Fixes panic when Parameters aren't defined.

File: backend/config.go [hunk 2]
```diff
@@ -86,6 +95,14 @@ func (s *configStore) resolveStack(path string, st stackConfig) stackConfig {
...
+		}
+
+		if stack.TemplateName == "" {
+			stack.TemplateName = stack.Name
```
Comment (kainosnoema) [inline backend/config.go:103]: This is documented in the readme, but wasn't working....

File: backend/fetcher.go [hunk 3]
```diff
@@ -38,15 +38,28 @@ func newFetcher(cs ConfigStore, ts TemplateStore, r ParamsResolver) *fetcher {
...
+func (f *fetcher) FetchAll() ([]client.Stack, error) {
+	stackConfigs, err := f.cs.FetchAll()
+	if err != nil {
+		return []client.Stack{}, fmt.Errorf("unable to fetch stacks: %s", err)
+	}
+
+	return f.fetchTemplates(stackConfigs)
+}
+
 func (f *fetcher) Fetch(name string) ([]client.Stack, error) {
```
Comment (defnotpaul) [inline backend/fetcher.go]: Should we maybe just update these to return a `client.StackList`?

Comment (kainosnoema) [reply backend/fetcher.go]: Yes probably. Didn’t know if it was a CLI concern or not but doesn’t hurt.

**Comment by defnotpaul** [comment]
I dig this. I was thinking about implementing a `list` feature but got hung up on the same questions you're having. I think supplying a flag for `--remote` or `--local` might make sense if we want to support both?

**Comment by kainosnoema** [comment]
@eyeamera I’ve made some changes locally to support showing remote stacks and their status too. Adds a bit of code that I want to clean up though. Thoughts?

**Comment by kainosnoema** [comment]
Ah missed your comment. Sounds good, I’ll work something up.

File: backend/fetcher.go (hunk 3 above)
Comment (defnotpaul) [reply backend/fetcher.go]: Hmmm, yeah, that's a good point. I think for now it makes sense, if only for the convenience.

Really though, I think we probably should be returning concrete types here.

**Comment by kainosnoema** [comment]
@eyeamera ok this is updated to use a string option and I've moved the common types into their own package.

**Comment by defnotpaul** [comment]
Sweet, this is awesome!

**Pull request merged**